File: RenderServer/src/main/java/me/hydos/alchemytools/network/packets/Packet.java
package me.hydos.alchemytools.network.packets;

import me.hydos.alchemytools.network.netty.NetworkBuffer;
import me.hydos.alchemytools.network.RenderThread;

public interface Packet {

    /**
     * Called inside the network thread and should be quick conversion of the data to bytes for sending to the client
     */
    default void write(NetworkBuffer buf) {
        throw new RuntimeException("Not Implemented");
    }

    /**
     * Called outside the Network Thread for handling incoming packets from the client
     */
    void handle(RenderThread renderThread);
}

File: RenderClient/packets/Packet.cs
namespace RenderClient; 

public interface Packet {
    
    /**
     * Called inside the network thread and should be quick conversion of the data to bytes for sending to the server
     */
    void Write(BinaryWriter writer);

    /**
     * Called outside the Network Thread for handling incoming packets from the server
     */
    void Handle();
}